book/src/custom-log-output.md

```diff
@@ -0,0 +1,186 @@
+# Customizing the log output
+
+The way a printer outputs logs can be customized by providing a format string to `defmt`. The format string takes a set of metadata and format specifiers which can be used to include or exclude certain information when printing the logs.
+
+## Basics
+
+> The following log will be used as reference in the examples below: `defmt::debug!("hello");`
+
+The simplest format string is `"{s}"`. This prints the log and nothing else:
+
+```text
+hello
+```
+
+Arbitrary text can be added to the format string, which will be printed as specified with each log.
+For example, `"Log: {s}"`:
+
+```text
+Log: hello
+```
+
+Multiple specifiers can be included within a format string, in any order. For example `"[{L}] {s}"` prints:
+
+```text
+[DEBUG] hello
+```
+
+## Metadata specifiers
+
+There are several metadata specifiers available that can be used in a format string.
+
+#### Log - `{s}`
+
+This specifier prints the actual log contents. For `defmt::info!("hello");`, this specifier prints `hello`.
+
+#### Crate name - `{c}`
+
+This specifier prints the name of the crate where the log is coming from.
+
+#### File name - `{f}`
+
+For a log coming from a file `/path/to/crate/src/foo/bar.rs`, this specifier prints `bar.rs`.
+
+This specifier can be used to print more detailed parts of the file path. The number of `f`s in the specifier determines how many levels up the path should be printed. For example, `{ff}` prints `foo/bar.rs`, and `{fff}` prints `src/foo/bar.rs`.
+
+#### File path - `{F}`
+
+For a log coming from a file `/path/to/crate/src/foo/bar.rs`, this specifier prints `/path/to/crate/src/foo/bar.rs`.
+
+#### Line number - `{l}`
+
+This specifier prints the line number where the log is coming from.
+
+#### Log level - `{L}`
+
+This specifier prints the log level. The log level is padded to 5 characters by default, for alignment purposes. For `defmt::info!("hello);`, this prints `INFO `.
+
+#### Module path - `{m}`
+
+This specifier prints the module path of the function where this log is coming from. This prints `my_crate::foo::bar` for the log shown below:
+
+```ignore
+// crate: my_crate
+mod foo {
+    fn bar() {
+        defmt::info!("hello");
+    }
+}
+```
+
+#### Timestamp - `{t}`
```

Comment:
I think #789 means we need some more entries in here now?

Reply:
Not really, the `{t}` specifier prints whatever `defmt::timestamp!` prints